internal/core/window/popup.rs

```diff
@@ -0,0 +1,333 @@
+// Copyright © SixtyFPS GmbH <info@slint.dev>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-Royalty-free-2.0 OR LicenseRef-Slint-Software-3.0
+
+//! Pupup window handling helpers
+
+use crate::lengths::{LogicalPoint, LogicalRect, LogicalSize};
+
+/// A collection of data that might influence the palcement of a `Popup`.
+pub enum Placement {
+    /// Request a fixed position
+    Fixed(LogicalRect),
+}
+
+/// Find a placement for the `Popup`, using the provided `Placement`.
+/// When a `clip_region` is provided, then the `Popup` will stay within those bounds.
+/// The `clip_region` typically is the window or the screen the window is on.
+pub fn place_popup(placement: Placement, clip_region: &Option<LogicalRect>) -> Option<LogicalRect> {
```

Comment:
What does it mean for this function to return None ?

Reply:
But what can be done if the popup doesn't fit?
Not showing it at all doesn't seem to be the solution.

Reply:
We can have it render centered on its Window maybe?

That might cut off UI elements necessary to dismiss the popup again though, so that is not good either.